src/Microsoft.AspNetCore.Razor.Language/Intermediate/IntermediateNodeBuilder.cs

```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.AspNetCore.Razor.Language.Intermediate
 {
-    public abstract class IntermediateNodeBuilder
+    internal abstract class IntermediateNodeBuilder
```

Comment:
@rynowak do you envision us ever wanting to expose this? I'm thinking no.

If so, no need for the abstractness and a default implementation (only one implementation); far simpler to just have all the logic live here.